### PR TITLE
Refactor results.php

### PR DESCRIPTION
**Pull request by kamil-tekiela**
This PR is a little chaotic as I wasn't very organized when refactoring this class. The goal was to reduce complexity. I removed redundant code, checks, temporary variables and unnecessary parameters. I added type hints, type declarations, stricter conditions, early returns, etc. 

This PR contains 2 bug fixes:
- Bug #17239, which should be fixed in full. I wasn't able to determine if one of the existing unit tests should cover this. Can someone suggest which test to modify?
- Undocumented bug related to sorting functionality. When using dynamic column names (e.g. ``SELECT CONCAT(table.col1, table1.`col2`)``) the sorting wouldn't work. The logic still seems flaky, but at least that small inconsistency is fixed. 

I have removed some methods that didn't look useful at all. I have completely redesigned or changed the signatures of a few other methods. 

I decided to remove unnecessary whitespace from HTML. HTML ignores contiguous whitespace. This should reduce slightly the length of the HTML output. I had to adjust unit tests to account for the removed whitespace. A lot of the functionality involved in the generation of CSS classes was outdated and no longer required. 

While the goal wasn't to reduce the length of code, this PR reduced it by 157 lines. It also got rid of many Psalm and PHPStan issues. I tested it by running PHPUnit locally and performing UAT myself. I haven't noticed any accidental regressions. 

P.S. This will probably be the last one of the small refactorings for some time. I will try to spend some time on more meaningful fixes or refactorings. For example, the master branch contains few issues related to PHP 8.1 changes. I will try to hunt them down and fix them.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Merged. Thank you!

File: js/src/makegrid.js [hunk 1]
```diff
@@ -1725,7 +1725,7 @@ var makeGrid = function (t, enableResize, enableReorder, enableVisib, enableGrid
 
             // make sure we have more than one column
             if ($firstRowCols.length > 1) {
-                var $colVisibTh = $(g.t).find('th:not(.draggable)');
+                var $colVisibTh = $(g.t).find('th:not(.draggable)').slice(0, 1);
```
Comment (williamdes) [inline js/src/makegrid.js:1728]: I do not understand what does this fix change, can you help me ?

Comment (kamil-tekiela) [reply js/src/makegrid.js:1728]: When repeating headers are switched on the column selections buttons would appear in the wrong place. To overcome this, I decided to keep them only in the first header.

Comment (williamdes) [reply js/src/makegrid.js:1728]: Thanks, for the record here is a screenshot
image

**Pull request merged**